From foo@baz Wed Jul 15 11:06:01 PDT 2009
Date: Wed, 15 Jul 2009 11:06:01 -0700
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: remove PAGE_SIZE and PAGE_SHIFT and __builtin  functions

From: Greg Kroah-Hartman <gregkh@suse.de>

The kernel provides all of this, and actually gets it correct, so don't
try to redefine these types of things.

Cc: Hank Janssen <hjanssen@microsoft.com>
Cc: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/BlkVsc.c      |    1 +
 drivers/staging/hv/NetVsc.c      |    1 +
 drivers/staging/hv/RndisFilter.c |    1 +
 drivers/staging/hv/StorVsc.c     |    2 ++
 drivers/staging/hv/Vmbus.c       |    1 +
 drivers/staging/hv/include/osd.h |   28 ----------------------------
 6 files changed, 6 insertions(+), 28 deletions(-)

--- a/drivers/staging/hv/BlkVsc.c
+++ b/drivers/staging/hv/BlkVsc.c
@@ -21,6 +21,7 @@
  */
 
 #include <linux/kernel.h>
+#include <linux/mm.h>
 #include "StorVsc.c"
 
 static const char* gBlkDriverName="blkvsc";
--- a/drivers/staging/hv/include/osd.h
+++ b/drivers/staging/hv/include/osd.h
@@ -29,34 +29,6 @@
 // Defines
 //
 
-#ifndef PAGE_SIZE
-#define PAGE_SIZE 0x1000
-#endif
-
-#ifndef PAGE_SHIFT
-#define PAGE_SHIFT 12
-#endif
-
-#ifndef memcpy
-#define memcpy		__builtin_memcpy
-#endif
-
-#ifndef memset
-#define memset		__builtin_memset
-#endif
-
-#ifndef memcmp
-#define memcmp		__builtin_memcmp
-#endif
-
-#ifndef strcpy
-#define strcpy		__builtin_strcpy
-#endif
-
-//
-//#ifndef sprintf
-//#define sprintf				__builtin_sprintf
-//#endif
 
 #define STRUCT_PACKED		__attribute__((__packed__))
 #define STRUCT_ALIGNED(x)	__attribute__((__aligned__(x)))
--- a/drivers/staging/hv/NetVsc.c
+++ b/drivers/staging/hv/NetVsc.c
@@ -23,6 +23,7 @@
 #define KERNEL_2_6_27
 
 #include <linux/kernel.h>
+#include <linux/mm.h>
 #include "include/logging.h"
 #include "NetVsc.h"
 #include "RndisFilter.h"
--- a/drivers/staging/hv/RndisFilter.c
+++ b/drivers/staging/hv/RndisFilter.c
@@ -24,6 +24,7 @@
 #define KERNEL_2_6_27
 
 #include <linux/kernel.h>
+#include <linux/mm.h>
 #include "include/logging.h"
 
 #include "include/NetVscApi.h"
--- a/drivers/staging/hv/StorVsc.c
+++ b/drivers/staging/hv/StorVsc.c
@@ -24,6 +24,8 @@
 #define KERNEL_2_6_27
 
 #include <linux/kernel.h>
+#include <linux/string.h>
+#include <linux/mm.h>
 #include "include/logging.h"
 
 #include "include/StorVscApi.h"
--- a/drivers/staging/hv/Vmbus.c
+++ b/drivers/staging/hv/Vmbus.c
@@ -22,6 +22,7 @@
  */
 
 #include <linux/kernel.h>
+#include <linux/mm.h>
 #include "include/logging.h"
 #include "VersionInfo.h"
 #include "VmbusPrivate.h"
